applications/contrastive_phenotyping/evaluation/cosine_similarity.py

```diff
@@ -0,0 +1,546 @@
+# %%
+# Import necessary libraries, try euclidean distance for both features and
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from sklearn.preprocessing import StandardScaler
+from umap import UMAP
+
+from viscy.representation.embedding_writer import read_embedding_dataset
+from viscy.representation.evaluation import (
+    calculate_cosine_similarity_cell,
+    compute_displacement,
+    compute_displacement_mean_std,
+)
+
+# %% Paths and parameters.
+
+
+features_path_30_min = Path(
+    "/hpc/projects/intracellular_dashboard/viral-sensor/infection_classification/models/time_sampling_strategies/time_interval/predict/feb_test_time_interval_1_epoch_178.zarr"
+)
+
+
+feature_path_no_track = Path(
+    "/hpc/projects/intracellular_dashboard/viral-sensor/infection_classification/models/time_sampling_strategies/negpair_random_sampling2/feb_fixed_test_predict.zarr"
+)
+
+
+features_path_any_time = Path(
+    "/hpc/projects/intracellular_dashboard/viral-sensor/infection_classification/models/time_sampling_strategies/negpair_difcell_randomtime_sampling/Ver2_updateTracking_refineModel/predictions/Feb_2chan_128patch_32projDim/2chan_128patch_56ckpt_FebTest.zarr"
+)
+
+
+data_path = Path(
+    "/hpc/projects/intracellular_dashboard/viral-sensor/2024_02_04_A549_DENV_ZIKV_timelapse/8-train-test-split/registered_test.zarr"
+)
+
+
+tracks_path = Path(
+    "/hpc/projects/intracellular_dashboard/viral-sensor/2024_02_04_A549_DENV_ZIKV_timelapse/8-train-test-split/track_test.zarr"
+)
+
+
+# %% Load embedding datasets for all three sampling
+fov_name = "/B/4/6"
+track_id = 52
+
+embedding_dataset_30_min = read_embedding_dataset(features_path_30_min)
+embedding_dataset_no_track = read_embedding_dataset(feature_path_no_track)
+embedding_dataset_any_time = read_embedding_dataset(features_path_any_time)
+
```

Comment:
It is better to compute UMAP and PCA here using `compute_umap` and `compute_pca` imported above. And add to the dataset.